crates/red_knot_python_semantic/src/semantic_index/definition.rs

```diff
@@ -749,19 +754,40 @@ impl MatchPatternDefinitionKind {
 }
 
 #[derive(Clone, Debug)]
-pub struct ComprehensionDefinitionKind {
+pub struct ComprehensionDefinitionKind<'db> {
+    target_kind: TargetKind<'db>,
     iterable: AstNodeRef<ast::Expr>,
-    target: AstNodeRef<ast::ExprName>,
+    target: AstNodeRef<ast::Expr>,
     first: bool,
     is_async: bool,
 }
 
-impl ComprehensionDefinitionKind {
+impl<'db> ComprehensionDefinitionKind<'db> {
+    pub(crate) fn new(
+        target_kind: TargetKind<'db>,
+        iterable: AstNodeRef<ast::Expr>,
+        target: AstNodeRef<ast::Expr>,
+        is_first: bool,
+        is_async: bool,
+    ) -> Self {
+        Self {
+            target_kind,
+            iterable,
+            target,
+            first: is_first,
+            is_async,
+        }
+    }
```

Comment:
nit: I'd usually avoid adding a `new` constructor method if it's mirroring the exact fields that the struct contains similar to how it's being done for constructing `*DefinitionNodeRef`